plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1088,6 +1103,125 @@ def run(
             f"{self.fract_deflected*100}%"
         )
 
+        # Simulation has not run, because creating new particles changes the simulation
+        self._has_run = True
+
+        return self.results_dict
```

Comment:
If I understand correctly, `results_dict` is really only used if someone wants to save the results, otherwise, the user accesses results via the `Tracker` attributes.  If I'm correct in this interpretation, then I wouldn't return `results_dict` here.  If I'm wrong, then I'd rename `results_dict` -> `results` and advertise it as the main way to interface with simulation results.

Reply:
Currently the workflow is set up so the user can also directly pass a `Tracker` object into `synthetic_radiograph`. But, I agree that this isn't necessary here in either case, so I'll remove it.